crates/agglayer-clock/src/block.rs

```diff
@@ -302,26 +306,37 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn test_block_clock_starting_with_wrong_genesis() {
+    async fn test_block_clock_with_genesis_in_future() {
+        let anvil = Anvil::new().block_time(1u64).spawn();
+        let client = Provider::<Ws>::connect(anvil.ws_endpoint()).await.unwrap();
+
+        let clock = BlockClock::new(client, 2, NonZeroU64::new(3).unwrap());
```

Comment:
I'd either increase the block time or move the genesis slightly more in the future than block 2, otherwise with a block time of 1s for anvil, it feels that the genesis block might not be in the future once spawning the block clock no?